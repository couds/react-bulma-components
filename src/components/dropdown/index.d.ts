import * as React from 'react';
import { ElementProps, RenderAsComponent } from '..';
import { Color } from '..';

interface DropdownProps<T> {
  value?: T;
  onChange?: (newValue: T) => void;
  color?: Color;
  hoverable?: boolean;
  label?: React.ReactNode;
  closeOnSelect?: boolean;
  right?: boolean;
  up?: boolean;
  align?: 'right';
  icon?: React.ReactNode;
  disabled?: boolean;
}

interface DropdownItemProps<T> {
  value: T;
}

declare const Dropdown: (<TValue, TComponent extends RenderAsComponent = 'div'>(
  props: DropdownProps<TValue> &
    ElementProps<DropdownProps<TValue>, TComponent>,
) => React.ReactElement) & {
  Item: <TValue, TComponent extends RenderAsComponent = 'div'>(
    props: DropdownItemProps<TValue> &
      ElementProps<DropdownItemProps<TValue>, TComponent>,
  ) => React.ReactElement;
  Divider: (
    props: Omit<ElementProps<{}, 'hr'>, 'renderAs'>,
  ) => React.ReactElement;
};

export default Dropdown;