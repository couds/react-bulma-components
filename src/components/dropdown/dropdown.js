import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import CONSTANTS from '../../constants';
import DropdownItem from './components/item';
import DropdownDivider from './components/divider';
import Button from '../button';
import Icon from '../icon';
import modifiers from '../../modifiers';
import Element from '../element';

const colors = [null].concat(Object.values(CONSTANTS.COLORS));

export default class Dropdown extends PureComponent {
  constructor(props) {
    super(props);
    this.domRef = props.domRef || React.createRef();
    this.state = {
      open: false,
    };
  }

  componentDidMount() {
    document.addEventListener('click', this.close);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.close);
  }

  close = evt => {
    // IDK yet how to test using the ref in enzime
    // istanbul ignore if
    if (
      this.props.hoverable ||
      (evt &&
        this.domRef &&
        this.domRef.current &&
        this.domRef.current.contains(evt.target))
    ) {
      return;
    }
    if (this.domRef.current) {
      this.setState({ open: false });
    }
  };

  toggle = evt => {
    if (this.props.hoverable) {
      return;
    }
    if (evt) {
      evt.preventDefault();
    }
    this.setState(({ open }) => ({ open: !open }));
  };

  select = value => () => {
    if (this.props.onChange) {
      this.props.onChange(value);
    }
    if (this.props.closeOnSelect) {
      this.close();
    }
  };

  render() {
    const {
      className,
      children,
      value,
      color,
      align,
      right,
      up,
      hoverable,
      label,
      onChange,
      closeOnSelect,
      ...props
    } = this.props;
    let current = label;

    const childrenArray = React.Children.map(children, (child, i) => {
      if (
        child.type === DropdownItem &&
        ((i === 0 && !label) || child.props.value === value)
      ) {
        current = child.props.children;
      }
      return React.cloneElement(
        child,
        child.type === DropdownItem
          ? {
              active: child.props.value === value,
              onClick: this.select(child.props.value),
            }
          : {},
      );
    });

    if (align === 'right') {
      // eslint-disable-next-line no-console
      console.warn(
        'react-bulma-components: "Align" prop will be replaced by "right" prop in future releases. Please update your code to avoid breaking changes.',
      );
    }

    return (
      <Element
        {...props}
        domRef={this.domRef}
        className={classnames('dropdown', className, {
          'is-active': this.state.open,
          'is-up': up,
          'is-right': right || align === 'right',
          'is-hoverable': hoverable,
        })}
      >
        <div
          className="dropdown-trigger"
          role="presentation"
          onClick={this.toggle}
        >
          <Button color={color}>
            <span>{current}</span>
            <Icon icon="angle-down" size="small" />
          </Button>
        </div>
        <div className="dropdown-menu" id="dropdown-menu" role="menu">
          <div className="dropdown-content">{childrenArray}</div>
        </div>
      </Element>
    );
  }
}

Dropdown.Item = DropdownItem;

Dropdown.Divider = DropdownDivider;

Dropdown.propTypes = {
  ...modifiers.propTypes,
  className: PropTypes.string,
  style: PropTypes.shape({}),
  children: PropTypes.node,
  value: PropTypes.any,
  onChange: PropTypes.func,
  color: PropTypes.oneOf(colors),
  right: PropTypes.bool,
  up: PropTypes.bool,
  align: PropTypes.oneOf(['right']),
  hoverable: PropTypes.bool,
  label: PropTypes.node,
  closeOnSelect: PropTypes.bool,
};

Dropdown.defaultProps = {
  ...modifiers.defaultProps,
  closeOnSelect: true,
  className: undefined,
  renderAs: 'div',
  domRef: undefined,
  style: undefined,
  value: undefined,
  children: [],
  onChange: undefined,
  color: undefined,
  align: undefined,
  hoverable: undefined,
  label: undefined,
};
