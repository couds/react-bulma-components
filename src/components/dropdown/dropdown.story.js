import React from 'react';

import { boolean, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Dropdown from '.';
import Container from '../container';
import Section from '../section';

const alignment = {
  Default: '',
  right: 'right',
};

class Wrapper extends React.Component {
  state = {
    selected: '',
  };
  onChange = selected => {
    this.setState({ selected });
  };
  render() {
    return (
      <Dropdown
        {...this.props}
        value={this.state.selected}
        onChange={this.onChange}
        color="info"
        label={text('label', '')}
      >
        <Dropdown.Item value="item">Dropdown item</Dropdown.Item>
        <Dropdown.Item value="other">Other Dropdown item</Dropdown.Item>
        <Dropdown.Item value="active">Active Dropdown item</Dropdown.Item>
        <Dropdown.Item value="other 2">Other Dropdown item</Dropdown.Item>
        <Dropdown.Divider />
        <Dropdown.Item value="divider">With divider</Dropdown.Item>
      </Dropdown>
    );
  }
}

storiesOf('Dropdown', module)
  .add('Default', () => (
    <Dropdown onChange={action('select')} label={text('label', '')}>
      <Dropdown.Item value="item">Dropdown item</Dropdown.Item>
      <Dropdown.Item value="other">Other Dropdown item</Dropdown.Item>
      <Dropdown.Item value="active">Active Dropdown item</Dropdown.Item>
      <Dropdown.Item value="other 2">Other Dropdown item</Dropdown.Item>
      <Dropdown.Divider />
      <Dropdown.Item value="divider">With divider</Dropdown.Item>
    </Dropdown>
  ))
  .add('Hoverable', () => (
    <Dropdown hoverable value="item" label={text('label', '')}>
      <Dropdown.Item value="item">Dropdown item</Dropdown.Item>
      <Dropdown.Item value="other">Other Dropdown item</Dropdown.Item>
      <Dropdown.Item value="active">Active Dropdown item</Dropdown.Item>
      <Dropdown.Item value="other 2">Other Dropdown item</Dropdown.Item>
      <Dropdown.Divider />
      <Dropdown.Item value="divider">With divider</Dropdown.Item>
    </Dropdown>
  ))
  .add('Alignment', () => (
    <div>
      <Container>
        <Section size="large">
          <Dropdown
            align={select(
              'align (deprecated; will be removed in future releases)',
              alignment,
            )}
            right={boolean('right (will replace "align" prop)', false)}
            value="item"
            up={boolean('up', false)}
            label={text('label', '')}
          >
            <Dropdown.Item value="item">Dropdown item</Dropdown.Item>
            <Dropdown.Item value="other">Other Dropdown item</Dropdown.Item>
            <Dropdown.Item value="active">Active Dropdown item</Dropdown.Item>
            <Dropdown.Item value="other 2">Other Dropdown item</Dropdown.Item>
            <Dropdown.Divider />
            <Dropdown.Item value="divider">With divider</Dropdown.Item>
          </Dropdown>
        </Section>
      </Container>
    </div>
  ))
  .add('Controlled component', () => <Wrapper />)
  .add('Controlled component Hoverable', () => (
    <Wrapper hoverable color="dark" />
  ));
