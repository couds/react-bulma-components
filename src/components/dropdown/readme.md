## Dropdown Component

An interactive dropdown menu for discoverable content ([Bulma docs](https://bulma.io/documentation/components/dropdown/))


### Usage:

- `import Dropdown from 'react-bulma-components/lib/components/dropdown'` to only import this components and his css (**Recommended**)
- `import { Dropdown } from react-bulma-components`

```jsx
<Dropdown>
  <Dropdown.Item value="item" >
    Dropdown item
  </Dropdown.Item>
  <Dropdown.Item value="other">
    Other Dropdown item
  </Dropdown.Item>
  <Dropdown.Item value="active">
    Active Dropdown item
  </Dropdown.Item>
  <Dropdown.Item value="other 2">
    Other Dropdown item
  </Dropdown.Item>
  <Dropdown.Divider />
  <Dropdown.Item value="divider">
    With divider
  </Dropdown.Item>
</Dropdown>
```

### Props

|property|propType|required|default|description|Bulma equivalent class|
|--- |--- |--- |--- |--- |--- |
|className|`string`|no|`''`|||
|style|`object`|no|`{}`|||
|value|`any`|no|`null`|Currently selected value of the dropdown||
|onChange|`func`|no|`null`|Function that will be called when the selected value change||
|color|`enum ['primary', 'success', 'info', 'warning', 'danger', 'light', 'dark', 'white', 'black', 'link']`|no|`null`|The color of dropdown using the classes from Bulma|`is-primary`, `is-success`, `is-info`, `is-warning`, `is-danger`, `is-light`, `is-dark`, `is-white`, `is-black`, `is-link`|
|align|`enum`|no|`null`|Alignment of the Dropdown, by default is Left|`is-left`, `is-right`|
|hoverable|`bool`|no|`false`|True if you want to open the dropdown on mouse hover|`hoverable`|

