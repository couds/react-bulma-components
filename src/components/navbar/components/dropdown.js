import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import modifiers from '../../../modifiers';
import Element from '../../element';
import renderAsShape from '../../../modifiers/render-as';

const NavbarDropdown = ({ className, boxed, right, children, ...props }) => (
  <Element
    {...props}
    className={classnames('navbar-dropdown', className, {
      'is-boxed': boxed,
      'is-right': right,
    })}
  >
    {children}
  </Element>
);

NavbarDropdown.propTypes = {
  ...modifiers.propTypes,
  style: PropTypes.shape({}),
  className: PropTypes.string,
  children: PropTypes.node,
  renderAs: renderAsShape,
  boxed: PropTypes.bool,
  right: PropTypes.bool,
};

NavbarDropdown.defaultProps = {
  ...modifiers.defaultProps,
  className: undefined,
  children: null,
  renderAs: 'span',
  boxed: false,
  right: false,
};

export default NavbarDropdown;
