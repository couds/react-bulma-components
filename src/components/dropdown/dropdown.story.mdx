import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { Dropdown } from '../..';
import * as stories from './dropdown.story';
import CommonProps from '../../../.storybook/common-props';

<Meta title="Components/Dropdown" />

# Dropdown

A component for implementing dropdown menu patterns that incorporates the button and the menu elements.

## Components

The `Dropdown` component contains the following components:

- `<Dropdown.Item />`: Defines an item in the dropdown menu.
- `<Dropdown.Divider />`: Draws a divider in the dropdown menu.

## Props

<br />

### `Dropdown`

<ArgsTable of={Dropdown} />

### `Dropdown.Item`

<ArgsTable of={Dropdown.Item} />

### `Dropdown.Divider`

<ArgsTable of={Dropdown.Divider} />

<CommonProps />

## Example

<Canvas>
  <Story story={stories.Overview} name="Dropdown" />
</Canvas>

## Related

- [Official documentation](https://bulma.io/documentation/components/dropdown/)