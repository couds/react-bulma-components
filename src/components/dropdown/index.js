import './dropdown.sass';

export * from './dropdown';
