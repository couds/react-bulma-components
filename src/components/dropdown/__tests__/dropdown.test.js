import { JSDOM } from 'jsdom';
import React from 'react';
import { fireEvent, render } from 'react-testing-library';
import { Dropdown } from '../dropdown';

describe('Dropdown component', () => {
  beforeEach(() => {
    global.window = new JSDOM('<body><div id="app-root"></div></body>').window;
  });
  it('Should have dropdown classname', () => {
    const { asFragment } = render(
      <Dropdown value="value" onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should add listener do document on mount', () => {
    global.window.document.addEventListener = jest.fn();
    render(
      <Dropdown value="value" onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(window.document.addEventListener).toHaveBeenCalled();
  });
  it('Should concat Bulma class with classes in props', () => {
    const { asFragment } = render(
      <Dropdown value="value" className="other-class" onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const { asFragment } = render(
      <Dropdown value="value" style={{ width: 400 }} onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should have divider', () => {
    const { asFragment } = render(
      <Dropdown value="value" style={{ width: 400 }} onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
        <Dropdown.Divider />
        <Dropdown.Item value="other">Other</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should be right-aligned when using "right" prop', () => {
    const { asFragment } = render(
      <Dropdown right>
        <Dropdown.Item value="value">Item</Dropdown.Item>
        <Dropdown.Item value="other">Other</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should appear above the dropdown button', () => {
    const { asFragment } = render(
      <Dropdown up>
        <Dropdown.Item value="value">Item</Dropdown.Item>
        <Dropdown.Item value="other">Other</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should open the Dropdown', () => {
    const { getByTestId } = render(
      <Dropdown value="value" style={{ width: 400 }} onChange={() => {}}>
        <Dropdown.Item value="value">Item</Dropdown.Item>
        <Dropdown.Divider />
        <Dropdown.Item value="other">Other</Dropdown.Item>
      </Dropdown>
    );
    expect(getByTestId('dropdown-container')).not.toHaveClass('is-active');
    fireEvent.click(getByTestId('dropdown-trigger'));
    expect(getByTestId('dropdown-container')).toHaveClass('is-active');
  });
  it('Should select the new value', async () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <Dropdown hoverable value="" style={{ width: 400 }} onChange={onChange}>
        <Dropdown.Item data-testid="foo" value="value">
          Foo
        </Dropdown.Item>
      </Dropdown>
    );
    fireEvent.mouseOver(getByTestId('dropdown-trigger'));
    fireEvent.click(getByTestId('foo'));
    expect(onChange).toHaveBeenCalledWith('value');
  });
  it('Should select the new value and close the dropdown', async () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <Dropdown value="" style={{ width: 400 }} onChange={onChange}>
        <Dropdown.Item data-testid="foo" value="value">
          Foo
        </Dropdown.Item>
      </Dropdown>
    );
    fireEvent.click(getByTestId('dropdown-trigger'));
    fireEvent.click(getByTestId('foo'));
    expect(onChange).toHaveBeenCalledWith('value');
    expect(getByTestId('dropdown-container')).not.toHaveClass('is-active');
  });
  it('Should show custom label passed to the label prop', () => {
    const { asFragment } = render(
      <Dropdown label="test label">
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should show the label of the first dropdown item when no custom label is passed', () => {
    const { asFragment } = render(
      <Dropdown>
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should show custom label when active valued is undefined/empty', () => {
    const { getByTestId } = render(
      <Dropdown label="test label" value="">
        <Dropdown.Item value="value">Item</Dropdown.Item>
      </Dropdown>
    );
    expect(getByTestId('dropdown-trigger')).toHaveTextContent('test label');
  });
  it('Should show the label of the dropdown item when value of it is the active value', () => {
    const { getByTestId } = render(
      <Dropdown label="test label" value="value">
        <Dropdown.Item value="value">Bar</Dropdown.Item>
      </Dropdown>
    );
    expect(getByTestId('dropdown-trigger')).toHaveTextContent('Bar');
  });
});
