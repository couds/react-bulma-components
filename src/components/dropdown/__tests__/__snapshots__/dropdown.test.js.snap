// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should appear above the dropdown button 1`] = `
<DocumentFragment>
  <div
    class="dropdown is-up"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item"
          role="presentation"
          title="value"
        >
          Item
        </div>
        <div
          class="dropdown-item"
          role="presentation"
          title="other"
        >
          Other
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should be right-aligned when using "right" prop 1`] = `
<DocumentFragment>
  <div
    class="dropdown is-right"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item"
          role="presentation"
          title="value"
        >
          Item
        </div>
        <div
          class="dropdown-item"
          role="presentation"
          title="other"
        >
          Other
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="dropdown other-class"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item is-active"
          role="presentation"
          title="value"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should have custom inline styles 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
    data-testid="dropdown-container"
    style="width: 400px;"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item is-active"
          role="presentation"
          title="value"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should have divider 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
    data-testid="dropdown-container"
    style="width: 400px;"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item is-active"
          role="presentation"
          title="value"
        >
          Item
        </div>
        <hr
          class="dropdown-divider"
        />
        <div
          class="dropdown-item"
          role="presentation"
          title="other"
        >
          Other
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item is-active"
          role="presentation"
          title="value"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should show custom label passed to the label prop 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          test label
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item"
          role="presentation"
          title="value"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component should show the label of the first dropdown item when no custom label is passed 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
    data-testid="dropdown-container"
  >
    <div
      data-testid="dropdown-trigger"
      role="presentation"
    >
      <button
        class="button"
        tabindex="0"
      >
        <span>
          Item
        </span>
        <span
          class="icon"
        >
          <i
            class="mdi mdi-chevron-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu"
      id="dropdown-menu"
      role="menu"
    >
      <div
        class="dropdown-content"
      >
        <div
          class="dropdown-item"
          role="presentation"
          title="value"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
